### PR TITLE
Fix issue #5409.

### PR DESCRIPTION
**Pull request by mfurkanuslu**
Fixes given issue by initializing Song and StepRecorder's uninitialized members in ctors.

### CODE REVIEW

File: src/gui/editors/PianoRoll.cpp [hunk 1]
```diff
@@ -152,6 +152,8 @@ PianoRoll::PianoRoll() :
 	m_zoomingModel(),
 	m_quantizeModel(),
 	m_noteLenModel(),
+	m_scaleModel(),
```
Comment (JohannesLorenz) [inline src/gui/editors/PianoRoll.cpp]: Isn't the default CTOR called anyways if there is no CTOR specified? In other words, do these two lines have any functional impact?

File: src/core/Song.cpp [hunk 2]
```diff
@@ -69,6 +69,7 @@ Song::Song() :
 	m_oldTicksPerBar( DefaultTicksPerBar ),
 	m_masterVolumeModel( 100, 0, 200, this, tr( "Master volume" ) ),
 	m_masterPitchModel( 0, -12, 12, this, tr( "Master pitch" ) ),
+	m_nLoadingTrack( 0 ),
```
Comment (JohannesLorenz) [inline src/core/Song.cpp]: The changes in `Song.cpp` look sane, I just wonder, is this related to the valgrind output in #5409?

File: src/gui/editors/PianoRoll.cpp (hunk 1 above)
Comment (mfurkanuslu) [reply src/gui/editors/PianoRoll.cpp]: I added those lines for consistency, there were other model members in member initialization list. Yes, it isn't related with this bug, but I didn't create separate commit because I think changes are similar.

File: src/core/Song.cpp (hunk 2 above)
Comment (mfurkanuslu) [reply src/core/Song.cpp]: Yes. I executed valgrind with  --track-origins=yes. It pointed a line that has new Song, expression which performs default initialization.

**Comment by JohannesLorenz** [comment]
@mfurkanuslu 

Your PR is OK, but can you please explain how you fixed that uninitialized read in `PianoRoll::PianoRoll() (PianoRoll.cpp:349)`? I'm just curious. :smile: 

Also, I'll rebase your PR to branch `stable-1.2` if you don't mind, that way we have the fix in both `stable-1.2` and `master` branches.

**Comment by mfurkanuslu** [comment]
Yes, rebase please.

**Comment by mfurkanuslu** [comment]
I don't know how it is fixed too :D.

**Comment by JohannesLorenz** [comment]
Many thanks for the commit @mfurkanuslu !

This will be merged to master now, and we'll cherry-pick relevant parts to `stable-1.2`. (Thanks to @PhysSong for the merge hints).
